builders/ethereum/dev-env/thirdweb.md

```diff
@@ -110,184 +112,206 @@ thirdweb offers SDKs for a range of programming languages, such as React, React
+
+This document will show you how to interact with your contract deployed to Moonbeam using React. You can view the [full React SDK reference](https://portal.thirdweb.com/typescript/v5/react){target=\_blank} in thirdweb’s documentation.
 
-This document will show you how to interact with your contract deployed to Moonbeam using React. You can view the [full React SDK reference](https://portal.thirdweb.com/react){target=\_blank} in thirdweb’s documentation.
+### Run Locally {: #run-locally }
 
-To create a new application pre-configured with thirdweb’s SDKs run and choose your preferred configurations:
+To run your dApp locally for testing and debugging purposes, use the command: 
 
 ```bash
-npx thirdweb create app --evm
+yarn dev
 ```
 
-Or install it into your existing project by running:
+The app will compile and specify the localhost and port number for you to visit in your browser.
 
-```bash
-npx thirdweb install
-```
+![thirdweb run locally](/images/builders/ethereum/dev-env/thirdweb/thirdweb-4.webp)
 
-### Initialize SDK On Moonbeam {: #initialize-sdk-on-moonbeam }
+### Create Client ID {: #create-client-id }
 
-Wrap your application in the `ThirdwebProvider` component and change the `activeChain` to Moonbeam.
+If you generated your thirdweb app with Vite, you'll have a `client.ts` file that looks like the below. As long you've created a `.env` file with your thirdweb API Key (Client ID) defined in `VITE_TEMPLATE_CLIENT_ID`, you can leave the `client.ts` as is and proceed to the next section.
 
-```javascript
-import { ThirdwebProvider } from '@thirdweb-dev/react';
-import { Moonbeam } from '@thirdweb-dev/chains';
+```typescript title="client.ts"
+import { createThirdwebClient } from "thirdweb";
+
+// Replace this with your client ID string
+// refer to https://portal.thirdweb.com/typescript/v5/client on how to get a client ID
+const clientId = import.meta.env.VITE_TEMPLATE_CLIENT_ID;
+
+export const client = createThirdwebClient({
+  clientId: clientId,
+});
 
-const App = () => {
-  return (
-    <ThirdwebProvider activeChain={Moonbeam}>
-      <YourApp />
-    </ThirdwebProvider>
-  );
-};
+```
+
+### Configure Chain {: #configure-chain }
+thirdweb offers a small number of chains from `@thirdweb/chains` and does not include Moonbeam networks in that list, so you'll need to specify the network details including chain ID and RPC URL. You can create a custom chain with [`defineChain`](https://portal.thirdweb.com/references/typescript/v5/defineChain){target=\_blank} as follows:
+
+```typescript title="App.tsx"
+import { defineChain } from "thirdweb";
+
+  const moonbase = defineChain({
+  id: BigInt(1287),
```

Comment:
I copied this into my project and I'm getting "Type 'bigint' is not assignable to type 'number'.ts(2322)" - can you please double-check it?